include/ck/tensor_operation/gpu/device/tensor_layout.hpp

```diff
@@ -3,6 +3,16 @@
 
 #pragma once
 
+#ifndef PP_DEFINE_LAYOUT_TYPE
```

Comment:
what does ```PP_``` mean?

Reply:
As discussed, we should remove those macro usages.

Reply:
Why? They made the code actually pretty succinct.

Reply:
We prefer using generator (like script), rather than using macro.  And we might change the layout type design in the future, so it would be better not to change rest of this file.